docs/architecture/modern-web-apps-azure/develop-asp-net-core-mvc-apps.md

```diff
@@ -92,6 +93,72 @@ For web APIs, ASP.NET Core MVC supports [_content negotiation_](/aspnet/core/mvc
 
 Web API projects should consider using the `[ApiController]` attribute, which can be applied to individual controllers, to a base controller class, or to the entire assembly. This attribute adds automatic model validation checking and any action with an invalid model will return a BadRequest with the details of the validation errors. The attribute also requires all actions have an attribute route, rather than using a conventional route, and returns more detailed ProblemDetails information in response to errors.
 
+### Keeping controllers under control
+
+For page-based applications, Razor Pages do a great job of keeping controllers from getting too large. Each individual page is given its own files and classes dedicated just to its handler(s). Prior to the introduction of Razor Pages, many view-centric applications would have large controller classes responsible for many different actions and views. These classes would naturally grow to have many responsibilities and dependencies, making them harder to maintain. If you find your view-based controllers are growing too large, consider refactoring them to use Razor Pages, or introducing a pattern like mediator.
+
+The mediator design pattern is used to reduce coupling between classes while allowing communication between them. In ASP.NET Core MVC applications, this pattern is frequently employed to break up controllers into smaller pieces by using *handlers* to do the work of action methods. The popular [MediatR NuGet package](https://www.nuget.org/packages/MediatR/) is often used to accomplish this. When using this approach, instead of having to provide controllers with every dependency their set of actions requires, only an instance of the mediator is required. Each action then uses the mediator instance to send a message, which is processed by a handler. The handler is specific to a single action and thus only needs the dependencies required by that action. An example of a controller using MediatR is shown here:
```

Comment:
This statement does not read right _ When using this approach, instead of having to provide controllers with every dependency their set of actions requires, only an instance of the mediator is required._

Reply:
Fixed